### PR TITLE
Disallow using '.' in 'ob thunk update'

### PR DESCRIPTION
**Pull request by hSloan**

### CODE REVIEW

**Comment by tomsmalley** [comment]
Probably want to check the canonicalized path to catch other variations of `.`, like

**Comment by hSloan** [comment]
> Probably want to check the canonicalized path to catch other variations of `.`, like

Thanks for the heads up. The deed is done.

File: lib/command/src/Obelisk/Command/Thunk.hs [hunk 1]
```diff
@@ -380,17 +380,22 @@ createThunkWithLatest target s = do
...
-    Right c -> case c of
-      ThunkData_Packed t -> return (target, t)
-      ThunkData_Checkout _ -> failWith "cannot update an unpacked thunk"
-  let src = _thunkPtr_source ptr
-  rev <- getLatestRev src
-  overwriteThunk overwrite $ ThunkPtr
-    { _thunkPtr_source = src
-    , _thunkPtr_rev = rev
-    }
+  canonicalCurrentPath <- liftIO $ canonicalizePath "."
```
Comment (luigy) [inline lib/command/src/Obelisk/Command/Thunk.hs]: > Probably want to check the canonicalized path to catch other variations of `.`, like
> [obelisk/lib/command/src/Obelisk/Command/Thunk.hs](https://github.com/obsidiansystems/obelisk/blob/ee5acbb7ad30105e8e98041616d2c995890c87d9/lib/command/src/Obelisk/Command/Thunk.hs#L571)
> 
> Line 571 in [ee5acbb](/obsidiansystems/obelisk/commit/ee5acbb7ad30105e8e98041616d2c995890c87d9)
> 
>  checkThunkDirectory :: MonadObelisk m => FilePath -> m ()

Can this reuse that here + update the error msg to reflect that?

**Comment by 3noch** [comment]
Why would we want to disallow this completely? If the thunk is packed then updating on `.` should be harmless. Updating unpacked thunks probably isn't safe in general.

**Comment by luigy** [comment]
This is an improvement over the current broken behavior, but supporting `ob thunk update .` would be nice to have

**Comment by 3noch** [comment]
@luigy Agreed. Just seems like if we need to special case this anyway it might be no harder to just support it instead of prohibiting it.

**Pull request merged**